### PR TITLE
feat: adds Block Select Input

### PR DESCRIPTION
**Pull request by johnmcclumpha**
Adds the Block Select input

[note: previous display issues are "fixed" - or at least they appear to be working correctly]

### CODE REVIEW

**Comment by kevinleedrum** [comment]
The menu is still opening at the top for me as-is.  However if I remove a lot of the other content on the page, the menu placement works just fine, so maybe we can chalk it up to something weird with vuepress..?

image

**Comment by johnmcclumpha** [comment]
> The menu is still opening at the top for me as-is. However if I remove a lot of the other content on the page, the menu placement works just fine, so maybe we can chalk it up to something weird with vuepress..?

Yeah I'm hoping it is a bit of a vuepress thing - once we have it in situ it seems to work ok

**Review by mitcsutt — COMMENTED** [review]
Great tests. Just a little concerned about this
image

Seems instead of going under it goes to the side. Is that intended?

File: packages/mds/src/components.d.ts [hunk 1]
```diff
@@ -1527,6 +1552,20 @@ declare namespace LocalJSX {
...
         "value"?: string;
     }
+    interface MxBlockSelect {
+        "assistiveText"?: string;
+        "colspan"?: number;
+        "disabled"?: boolean;
+        "error"?: boolean;
+        "inputId"?: string;
+        "label"?: string;
+        "maxlength"?: number;
```
Comment (mitcsutt) [inline packages/mds/src/components.d.ts:1562]: You might need to run stencil build again as maxLength has been deleted since

**Comment by johnmcclumpha** [comment]
> Seems instead of going under it goes to the side. Is that intended?

yeah this is the same challenge that has it going above instead of below - it seems to vary greatly depending on window size and scroll position. part of me was contemplating removing it but that would change existing behaviour (and could also be problematic if it would potentially display off-screen)

I'm hoping that once it's in a better container (as in application instead of the magical section in vuepress) that it will work as intended (initial tests I've done seem to work)

**Review by kevinleedrum — APPROVED** [review]
At some point, we could look into migrating to [floating-ui](https://floating-ui.com/docs/migration), which is a spiritual successor to popper written by the same author, though I guess it could have the same issues.  In any case, I think it's ultimately a non-issue for now.  🚢

**Review by mitcsutt — APPROVED** [review]

**Pull request merged**